api/bin/chainflip-ingress-egress-tracker/src/main.rs

```diff
@@ -134,22 +260,142 @@ async fn start(
 	let (witness_sender, _) =
 		tokio::sync::broadcast::channel::<state_chain_runtime::RuntimeCall>(EVENT_BUFFER_SIZE);
 
-	witnessing::start(scope, settings, witness_sender.clone()).await?;
+	let (state_chain_client, env_params) =
+		witnessing::start(scope, settings, witness_sender.clone()).await?;
+	let btc_network = env_params.btc_network;
 
 	module.register_subscription(
 		"subscribe_witnessing",
 		"s_witnessing",
 		"unsubscribe_witnessing",
 		move |_params, mut sink, _context| {
 			let mut witness_receiver = witness_sender.subscribe();
+			let state_chain_client = state_chain_client.clone();
 
 			tokio::spawn(async move {
 				while let Ok(event) = witness_receiver.recv().await {
-					use codec::Encode;
-					if let Ok(false) = sink.send(&event.encode()) {
-						log::debug!("Subscription is closed");
-						break
+					use pallet_cf_broadcast::Call as BroadcastCall;
+					use pallet_cf_ingress_egress::Call as IngressEgressCall;
+					use state_chain_runtime::RuntimeCall::*;
+
+					macro_rules! send {
+						($value:expr) => {
+							if let Ok(false) = sink.send($value) {
+								log::debug!("Subscription is closed");
+								return
+							}
+						};
 					}
+
+					match event {
+						EthereumIngressEgress(IngressEgressCall::process_deposits {
+							deposit_witnesses,
+							block_height,
+						}) =>
+							for witness in deposit_witnesses as Vec<DepositWitness<Ethereum>> {
+								let info = WitnessInformation::from((witness, block_height));
+								send!(&info);
+							},
+						BitcoinIngressEgress(IngressEgressCall::process_deposits {
+							deposit_witnesses,
+							block_height,
+						}) =>
+							for witness in deposit_witnesses as Vec<DepositWitness<Bitcoin>> {
+								let info =
+									WitnessInformation::from((witness, block_height, btc_network));
+								send!(&info);
+							},
+						PolkadotIngressEgress(IngressEgressCall::process_deposits {
+							deposit_witnesses,
+							block_height,
+						}) =>
+							for witness in deposit_witnesses as Vec<DepositWitness<Polkadot>> {
+								let info = WitnessInformation::from((witness, block_height));
+								send!(&info);
+							},
+						EthereumBroadcaster(BroadcastCall::transaction_succeeded {
+							tx_out_id,
+							..
+						}) => {
+							let broadcast_id =
+								get_broadcast_id::<Ethereum>(&state_chain_client, &tx_out_id).await;
```

Comment:
Maybe we want to log warn when this is None, since we don't really expect it to be None - if we're making these queries we expect them to be None

Reply:
Done